azurerm/internal/services/storage/storage_account_resource.go

```diff
@@ -1535,12 +1524,12 @@ func resourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) err
 		}
 
 		if !hasWriteLock && !doesntHavePermissions {
-			return fmt.Errorf("Error listing Keys for Storage Account %q (Resource Group %q): %s", name, resGroup, err)
+			return fmt.Errorf("Error listing Keys for Storage Account %q (Resource Group %q): %s", storageAccountName, resourceGroupName, err)
 		}
 	}
 
 	d.Set("name", resp.Name)
-	d.Set("resource_group_name", resGroup)
+	d.Set("resource_group_name", resourceGroupName)
```

Comment:
We'll also need to set the value of `allow_shared_key_access` into the state to be able to track it.

Given the API has a default value (true) but can return `null` when this value isn't explicitly configured, we'll need to default this within Terraform using:

```suggestion
	allowSharedKeyAccess := true
	if props.AllowSharedKeyAccess != nil {
		allowSharedKeyAccess = *props.AllowSharedKeyAccess
	}
	d.Set("allow_shared_key_access", allowSharedKeyAccess)
```

(within the `if props := response.AccountProperties; props != nil` statement below)